### ISSUE TITLE
Package directory created even cancelled the install

### ISSUE THREAD

**Issue by davidlatwe** [mottosso/rez-pipz#33]
### Problem (Minor)
The package's directory gets created (empty) even I said no to the following question.
```
Packages will be installed to C:\Users\david\rez\packages\local-install
After this operation, 7.84 mb will be used.
Do you want to continue? [Y/n]
```
### Cause
The directory gets created at this if-statement
https://github.com/mottosso/rez-pipz/blob/4389c6776feec59aadf72db3601e574e5bed5620/python/pipz/cli.py#L123

Which will return result from this
https://github.com/mottosso/rez-pipz/blob/4389c6776feec59aadf72db3601e574e5bed5620/python/pipz/pip.py#L196

Which ends up in rez (same as bleeding-rez)
[src/rezplugins/package_repository/filesystem.py#L604-L606](https://github.com/mottosso/bleeding-rez/blob/cb1c71c54f0486ec23e777e66f0bfb5aead7e52c/src/rezplugins/package_repository/filesystem.py#L604-L606)

```python
        try:
            os.makedirs(path)
        except OSError as e:
```
So it looks like this issue should be fixed in there 🤔

**Comment by mottosso** [comment]
Yes, this is a bummer. The reason we can't merely check ourselves whether a package exist (by e.g. querying for path existance) is because variants can exist in multiple places, even in a database or the like. So it's piggy backing on Rez's already-advanced lookup mechanism (this `variant.install`) to shoulder that responsibiity.

And yeah.. I think we're probably stuck with that. If I'm not misremembering, or if they haven't fixed it already (worth asking).
